### DISCUSSION TITLE
🆕  Dependency Review Action (Beta) Feedback - [Update: Released]

### DISCUSSION THREAD

**Discussion by courtneycl** [community/community#14413] [Code Security] [answered]
We have a new [Dependency Review GitHub Action](https://github.com/actions/dependency-review-action) that can help prevent you from accidentally introducing known vulnerabilities into your code. The action is supported by a new API endpoint that diffs the dependencies between any two revisions. It’s currently in Beta and we’re looking for your feedback! The action is available for all public repositories, as well as private repositories that have Github Advanced Security licensed. 

Once added to your repository, the action will: 
* Scan your pull requests for any dependency changes
* Raise an error if any dependencies have known vulnerabilities
* Direct you to the GitHub Advisory Database so you can see how to resolve it 

![dr-fail](https://user-images.githubusercontent.com/3474250/160143642-2bb10d32-57e1-4577-ace8-dc4f36b44da4.gif)

Note that the action will only block a pull request from being merged if the repository owner has required the check to pass before merging. 

For more information, please see [the docs](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review#dependency-review-enforcement).

### Feedback 

Your feedback will help shape how the action grows, so please let us know what you think below. We’re particularly interested in how you’d like to configure the action to determine when it fails and when it blocks a merge (by vulnerability severity, branches, etc). 

Thank you for trying the action — we’re excited to hear from you!

**Comment by billxinli** [comment]
What is the relationship of the sources of vulnerabilities between this and `npm audit`?

**Reply by ruxandrafed** [reply]
https://github.blog/2021-10-07-github-advisory-database-now-powers-npm-audit/

**Reply by dar00009** [reply]
> reply

#13499

**Comment by saltan505** [comment]
> لدينا إجراء [GitHub لمراجعة التبعية](https://github.com/actions/dependency-review-action) جديد يمكن أن يساعد في منعك من إدخال ثغرات أمنية معروفة عن طريق الخطأ في التعليمات البرمجية الخاصة بك. يتم دعم الإجراء من خلال نقطة نهاية API جديدة تختلف في التبعيات بين أي مراجعتين. إنه حاليًا في الإصدار التجريبي ونحن نبحث عن تعليقاتك! الإجراء متاح لجميع المستودعات العامة ، بالإضافة إلى المستودعات الخاصة التي تم ترخيص Github Advanced Security لها.
> 
> بمجرد إضافته إلى المستودع الخاص بك ، سيقوم الإجراء بما يلي:
> 
> * افحص طلبات السحب بحثًا عن أي تغييرات في التبعية
> * رفع خطأ إذا كان لدى أي تبعيات ثغرات أمنية معروفة
> * قم بتوجيهك إلى قاعدة بيانات GitHub الاستشارية حتى تتمكن من معرفة كيفية حلها
> 
> ![دكتور فشل](https://user-images.githubusercontent.com/3474250/160143642-2bb10d32-57e1-4577-ace8-dc4f36b44da4.gif)
> 
> لاحظ أن الإجراء لن يؤدي إلا إلى حظر دمج طلب السحب إذا طلب مالك المستودع اجتياز الشيك قبل الدمج.
> 
> لمزيد من المعلومات ، يرجى الاطلاع [على المستندات](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review#dependency-review-enforcement) .
> 
> ### ردود الفعل
> ستساعد ملاحظاتك في تشكيل كيفية تطور الإجراء ، لذا يرجى إخبارنا برأيك أدناه. نحن مهتمون بشكل خاص بكيفية تكوين الإجراء لتحديد متى يفشل ومتى يحظر الدمج (حسب شدة الثغرة الأمنية والفروع وما إلى ذلك).
> 
> نشكرك على تجربة الإجراء - يسعدنا أن نسمع منك!

**Reply by saltan505** [reply]
> > لدينا إجراء [GitHub لمراجعة التبعية](https://github.com/actions/dependency-review-action) جديد يمكن أن يساعد في منعك من إدخال ثغرات أمنية عن طريق الخطأ في التعليمات الخاصة بك. يتم إجراء التبعيات من جديد في نهاية API. وهذا ما يجب أن نفعله في وقتنا هذا! الأماكن المتاحة لجميع المستودعات العامة ، بالإضافة إلى المستودعات التي تم ترخيص Github Advanced Security لها.
> > ما يلي:
> > 
> > * افحص طلبات السحب بحثًا عن أي٪٪ في التبعية
> > * إرسال رسالة أمنية؟
> > * قم بتوجيهك إلى قاعدة بيانات GitHub المشورة حتى تتمكن من معرفة كيفية حلها
> > 
> > ![دكتور فشل](https://user-images.githubusercontent.com/3474250/160143642-2bb10d32-57e1-4577-ace8-dc4f36b44da4.gif)
> > لاحظ أن يؤدي إلى الانتظار إلا إلى حظر السحب.
> > الرجاء الاطلاع [على المستندات](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review#dependency-review-enforcement) .
> > ### ردود الفعل
> > ستظهر ملاحظاتك في كيفية تشكيل النموذج ، لذا يرجى إخبارنا بورأيك. نحن مهتمون بشكل خاص بكيفية تكوين تكوين الإجراء المحدد متى يفشل ومتى يحظر الدمج (حسب شدة الثغرة الأمنية والفروع وما إلى ذلك).
> > نشكرك على تجربة الإجراء - يسعدنا أن نسمع منك!

**Reply by 2386512** [reply]
Yes

**Comment by Tmoney97** [comment]
What?

**Comment by Dementu** [comment]
Dependency of droguri!

**Reply by Pakinayy** [reply]
phakhinaysaree@gmail.com

**Comment by Milo123459** [comment]
Could this potentially just be added into Dependabot, rather than using a GitHub Action? It could trigger whenever a lock file or manifest changes.

**Reply by courtneycl** [reply]
Yes! The action is a first pass to get the functionality out there; the future plan is for it to work without needing an action added to each repo.

**Reply by Milo123459** [reply]
Thanks for the information!

**Comment by hen818** [comment]
https://github.com/github/feedback/discussions/14413#discussioncomment-2619409

**Comment by dliao-tyro-admin** [comment]
It's not entirely related to the GitHub action but more the Dependency Review work in general.

Would there be any considerations of implementing a waiver process? A use case would be when a developer acknowledges a Dependabot alert and marks it as `No bandwidth to fix` or `Risk is tolerable`. This alert may then resurface itself after some period of time. The period of time is likely going to vary based on company to company. E.g. At mine, we go by severity where high would be 14 days and medium would be 30 days.

**Reply by courtneycl** [reply]
Thanks for your feedback here! 

We're not planning on implementing this kind of advanced functionality for the dependency review action, but we'll keep this feedback in mind as we work on a more first-party experience for Dependabot alerts on PRs. 

We did add a configuration to the action that allows you to pass GHSA IDs that you want the action to not alert on -- that might solve part of your problem with wanting to waive alerts on particular vulnerable dependencies. See `allow-ghsas` in the [configuration options](https://github.com/actions/dependency-review-action#configuration-options) for details.

**Comment by laughedelic** [comment]
Is this action meant to work with the Dependency Submission API?

I've tried submitting dependencies via API (using https://github.com/scalacenter/sbt-dependency-submission), they show up correctly in the Dependency Graph. Then I tried adding vulnerable dependencies and running Dependency Review action. Dependabot detected vulnerabilities and raised alerts (visible on the Security page), but the review action hasn't detected any changes.

This might not be a problem with the action, but then underlying [API endpoint](https://docs.github.com/en/rest/dependency-graph/dependency-review?apiVersion=2022-11-28#get-a-diff-of-the-dependencies-between-commits). I tried calling it manually, but didn't get any results. Is this a known limitation and is there any work planned to make the two parts work together?

**Reply by courtneycl** [reply]
Hi @laughedelic -- there is work here to get the action working with submitted dependencies. It should land in the next few weeks.

**Reply by laughedelic** [reply]
Awesome! Thanks for the quick response :octocat:

**Comment by ettaboyle** [comment, marked as answer]
#### Nov. 14, 2022 [Update] 
## ChangeLog Post: [Dependency review API is generally available with updates to dependency review enforcement](https://github.blog/changelog/2022-11-14-dependency-review-api-is-generally-available-with-updates-to-dependency-review-enforcement/)

> The [dependency review API](https://docs.github.com/en/rest/dependency-graph/dependency-review) is now generally available.
>
> The [Dependency Review GitHub Action](https://github.com/actions/dependency-review-action) now allows you to reference a local or external configuration file. There are also new configuration options:
> 
> - ` fail-on-scopes:` contains a list of strings representing the build environments you want to support (`development`, `runtime`, `unknown`). The action will fail on pull requests that introduce vulnerabilities in the scopes that match the list
> - `allow-ghsas`: contains a list of GitHub Security Advisory IDs that can be skipped during detection
> - `license-check` and `vulnerability-check`: a boolean option that allows you disable either one of the checks
> 
> Learn more about the [dependency graph](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph) and [dependency review](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review)
